src/coreclr/unwinder/loongarch64/unwinder_loongarch64.cpp

```diff
@@ -0,0 +1,1673 @@
+        proper type is associated with the function, then the address of the
+        language specific exception handler is returned. Otherwise, NULL is
+        returned.
+
+    HandlerData - Supplies a pointer to a variable that receives a pointer
+        the the language handler data.
+
+    UnwindParams - Additional parameters shared with caller.
+
+Return Value:
+
+    STATUS_SUCCESS if the unwind could be completed, a failure status otherwise.
+    Unwind can only fail when validation bounds are specified.
+
+--*/
+
+{
+    ULONG Count;
+    ULONG Cr;
+    ULONG CurrentOffset;
+    ULONG EpilogLength;
+    ULONG Flag;
+    ULONG FloatSize;
+    ULONG FrameSize;
+    ULONG FRegOpcodes;
+    ULONG FunctionLength;
+    ULONG HBit;
+    ULONG HOpcodes;
+    ULONG IRegOpcodes;
+    ULONG IntSize;
+    ULONG LocalSize;
+    DWORD64 OffsetInFunction;
+    DWORD64 OffsetInScope;
+    ULONG PrologLength;
+    ULONG RegF;
+    ULONG RegI;
+    ULONG RegSize;
+    ULONG ScopeStart;
+    ULONG StackAdjustOpcodes;
+    NTSTATUS Status;
+    ULONG UnwindData;
+
+    UnwindData = FunctionEntry->UnwindData;
+    Status = STATUS_SUCCESS;
+
+    //
+    // Compact records always describe an unwind to a call.
+    //
+
+    ContextRecord->ContextFlags |= CONTEXT_UNWOUND_TO_CALL;
+
+    //
+    // Extract the basic information about how to do a full unwind.
+    //
+
+    Flag = UnwindData & 3;
+    FunctionLength = (UnwindData >> 2) & 0x7ff;
+    RegF = (UnwindData >> 13) & 7;
+    RegI = (UnwindData >> 16) & 0xf;
+    HBit = (UnwindData >> 20) & 1;
+    Cr = (UnwindData >> 21) & 3;
+    FrameSize = (UnwindData >> 23) & 0x1ff;
+
+    assert(!"---------------LOONGARCH64 ShouldNotReachHere");
+    if (Flag == 3) {
+        return STATUS_UNWIND_INVALID_SEQUENCE;
+    }
+    if (Cr == 2) {
+        return STATUS_UNWIND_INVALID_SEQUENCE;
+    }
+
+    //
+    // Determine the size of the locals
+    //
+
+    IntSize = RegI * 8;
+    if (Cr == 1) {
+        IntSize += 8;
+    }
+    FloatSize = (RegF == 0) ? 0 : (RegF + 1) * 8;
+    RegSize = (IntSize + FloatSize + 8*8 * HBit + 0xf) & ~0xf;
+    if (RegSize > 16 * FrameSize) {
+        return STATUS_UNWIND_INVALID_SEQUENCE;
+    }
+    LocalSize = 16 * FrameSize - RegSize;
+
+    //
+    // If we're near the start of the function (within 17 words),
+    // see if we are within the prolog.
+    //
+    // N.B. If the low 2 bits of the UnwindData are 2, then we have
+    // no prolog.
+    //
+
+    OffsetInFunction = (ControlPcRva - FunctionEntry->BeginAddress) / 4;
+    OffsetInScope = 0;
+    if (OffsetInFunction < 17 && Flag != 2) {
+
+        //
+        // Compute sizes for each opcode in the prolog.
+        //
+
+        IRegOpcodes = (IntSize + 8) / 16;
+        FRegOpcodes = (FloatSize + 8) / 16;
+        HOpcodes = 4 * HBit;
+        StackAdjustOpcodes = (Cr == 3) ? 1 : 0;
+        if (Cr != 3 || LocalSize > 512) {///TODO for LOONGARCH64: 512 is smaller than real!
```

Comment:
Should this TODO be handled in this PR?

Reply:
OK, I will amend it.
Thanks

Reply:
I will just delete the comment `TODO`.
Here the `512` is just the JIT unwind info rather than the ISA's info.
Now the LoongArch64's implementation for JIT-unwind is referenced the arm64, althought we can redesign based on the LoongArch64's ISA which might be better.